test/e2e_env/universal/meshidentity/spire.go

```diff
@@ -0,0 +1,180 @@
+		Expect(NewClusterSetup().
+			Install(ResourceUniversal(samples.MeshDefaultBuilder().WithName(meshName).WithMeshServicesEnabled(v1alpha1.Mesh_MeshServices_Exclusive).Build())).
+			Install(DemoClientUniversal("demo-client", meshName,
+				WithTransparentProxy(true),
+				WithWorkload("demo-client"),
+				WithSpireAgent(demoClientToken, spireIP, strconv.FormatUint(uint64(spire.SpireServerPort), 10), trustDomain),
+			)).
+			Install(TestServerUniversal(testServerName, meshName,
+				WithArgs([]string{"echo", "--instance", "spire-test-server"}),
+				WithWorkload("test-server"),
+				WithSpireAgent(testServerToken, spireIP, strconv.FormatUint(uint64(spire.SpireServerPort), 10), trustDomain),
+			)).
+			Setup(universal.Cluster)).To(Succeed())
+
+		Expect(spireDeployment.RegisterWorkload(
+			universal.Cluster,
+			fmt.Sprintf("spiffe://%s/workload/demo-client", trustDomain),
+			fmt.Sprintf("spiffe://%s/agent/demo-client", trustDomain),
+			"unix:uid:5678",
+		)).To(Succeed())
+		Expect(spireDeployment.RegisterWorkload(
+			universal.Cluster,
+			fmt.Sprintf("spiffe://%s/workload/test-server", trustDomain),
+			fmt.Sprintf("spiffe://%s/agent/test-server", trustDomain),
+			"unix:uid:5678",
+		)).To(Succeed())
+	})
+
+	AfterEachFailure(func() {
+		DebugUniversal(universal.Cluster, meshName)
+	})
+
+	isMeshIdentityReady := func(cluster *UniversalCluster, name string) (bool, error) {
+		output, err := cluster.GetKumactlOptions().RunKumactlAndGetOutput("get", "meshidentity", "-m", meshName, name, "-o", "json")
+		if err != nil {
+			return false, err
+		}
+		return strings.Contains(output, "Successfully initialized"), nil
+	}
+
+	E2EAfterAll(func() {
+		Expect(universal.Cluster.DeleteMeshApps(meshName)).To(Succeed())
+		Expect(universal.Cluster.DeleteMesh(meshName)).To(Succeed())
+	})
+
+	It("should use identity from Spire for secure communication", func() {
+		mtp := fmt.Sprintf(`
+type: MeshTrafficPermission
+name: demo-client-to-test-server
+mesh: %s
+spec:
+  targetRef:
+    kind: Dataplane
+    labels:
+      kuma.io/workload: test-server
+  rules:
+  - default:
+      allow:
+      - spiffeID:
+          type: Exact
+          value: spiffe://%s/workload/demo-client
+`, meshName, trustDomain)
+		admin := universal.Cluster.GetApp(testServerName).GetEnvoyAdminTunnel()
+
+		// given
+		// communication works
+		Eventually(func(g Gomega) {
+			response, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", "test-server.svc.mesh.local",
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(response.Instance).To(Equal("spire-test-server"))
+		}, "30s", "1s").MustPassRepeatedly(5).Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			s, err := admin.GetStats("listener.*_80.ssl.handshake")
+			g.Expect(err).ToNot(HaveOccurred())
+			// tls is not enabled
+			g.Expect(s.Stats).To(BeEmpty())
+		}, "30s", "1s").Should(Succeed())
+
+		// when
+		// enable MeshIdentity and MeshTrafficPermissions
```

Comment:
Good test would be to enable MeshTrust/MeshIdentity and see the traffic doesn't work without MeshTrafficPermission